File: Second Largest/Second_Largest.py
class Solution:
    def getSecondLargest(self, arr):
        largest = float('-inf')
        second_largest = float('-inf')
        
        for i in arr:
            if i > largest:
                second_largest = largest
                largest = i
            elif i > second_largest and i < largest:
                second_largest = i
        
        if second_largest == float('-inf'):
            return -1
        return second_largest

File: Second Largest/Second_Largest.java
class Solution {
    public int getSecondLargest(int[] arr) {
        int largest = Integer.MIN_VALUE;
        int secondLargest = Integer.MIN_VALUE;

        for (int i : arr) {
            if (i > largest) {
                secondLargest = largest;
                largest = i;
            } else if (i > secondLargest && i < largest) {
                secondLargest = i;
            }
        }
        if (secondLargest == Integer.MIN_VALUE) return -1;
        return secondLargest;
    }
}